### PR TITLE
using reconnect interceptor in authorize.

### PR DESCRIPTION
**Pull request by harishdinne**

### CODE REVIEW

File: services/authorize/client.go [hunk 1]
```diff
@@ -113,8 +123,52 @@ type AuthorizeClient interface { // nolint: golint
 	RemoveUserRoleWithContext(ctx context.Context, roleName string) error
 }
 
+func getSecret(sess *session.Session, secretsName string, out interface{}) (err error) {
```
Comment (stroem) [inline services/authorize/client.go]: Refactor to new file? secret.go inside of this package. This file starting to get big

File: services/authorize/client.go [hunk 2]
```diff
@@ -113,8 +123,52 @@ type AuthorizeClient interface { // nolint: golint
...
+	// credentials - default
+	svc := secretsmanager.New(sess)
+	input := &secretsmanager.GetSecretValueInput{
+		SecretId: aws.String(secretsName),
+	}
+
+	result, err := svc.GetSecretValue(input)
+	if err != nil {
+		log.WithError(err).
```
Comment (stroem) [inline services/authorize/client.go]: Add WithTracing

File: services/authorize/client.go [hunk 3]
```diff
@@ -113,8 +123,52 @@ type AuthorizeClient interface { // nolint: golint
...
+		panic(err)
+	}
+
+	return grpc.WithTransportCredentialsPEM(
+		host,
+		clientCert.Crt, clientCert.Key, clientCert.CA,
+	)
+}
+
 type client struct {
```
Comment (stroem) [inline services/authorize/client.go]: Structs in the top

File: services/authorize/client.go [hunk 4]
```diff
@@ -125,24 +179,57 @@ func CreateClient() AuthorizeClient {
...
-func (c *client) DialWithContext(ctx context.Context, host, port string, opts ...grpc.DialOption) (err error) {
-	conn, err := grpc.DialContext(ctx, host+":"+port, opts...)
+func (c *client) DialWithContext(ctx context.Context, sess *session.Session, host, port, secretKey string, opts ...googleGrpc.DialOption) error {
+	reconnectOpts := googleGrpc.WithUnaryInterceptor(reconnect.UnaryInterceptor(
+		reconnect.WithCodes(codes.DeadlineExceeded),
+		reconnect.WithNewConnection(
+			func(_ context.Context, invokerConn *googleGrpc.ClientConn, invokerOptions ...googleGrpc.CallOption) (context.Context, *googleGrpc.ClientConn, []googleGrpc.CallOption, error) {
+				opt, err := getCredentialOption(sess, host, secretKey)
+				if err != nil {
+					return context.Background(), invokerConn, invokerOptions, err
```
Comment (stroem) [inline services/authorize/client.go]: Use the invokerCtx

File: services/authorize/client.go (hunk 2 above)
Comment (harishdinne) [reply services/authorize/client.go]: fixed

File: services/authorize/client.go (hunk 4 above)
Comment (harishdinne) [reply services/authorize/client.go]: fixed

File: services/authorize/client.go (hunk 1 above)
Comment (harishdinne) [reply services/authorize/client.go]: fixed

File: services/authorize/client.go (hunk 3 above)
Comment (harishdinne) [reply services/authorize/client.go]: fixed

**Pull request merged**